app/src/test/java/com/sun/training/ut/ui/exercise_two/ExerciseTwoViewModelTest.kt

```diff
@@ -0,0 +1,157 @@
+package com.sun.training.ut.ui.exercise_two
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.sun.training.ut.data.Constant
+import com.sun.training.ut.ui.exercise_one.ExerciseOneViewModel
+import io.mockk.spyk
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+
+class ExerciseTwoViewModelTest {
+
+    @Rule
+    @JvmField
+    val instantTaskExecutorRule: TestRule = InstantTaskExecutorRule()
+
+    private lateinit var viewModel: ExerciseTwoViewModel
+
+    @Before
+    fun setup() {
+        viewModel = spyk(ExerciseTwoViewModel())
+    }
+
+    @Test
+    fun onVipChecked_Case01() {
+        //Given
+        val isChecked = true
+
+        //When
+        viewModel.onVipChecked(isChecked)
+
+        //Then
+        Assert.assertEquals(isChecked, viewModel.isVip)
+    }
+
+    @Test
+    fun onTimeChanged_Case01() {
+        //Given
+        val hour = 1
+        val minute = 2
+
+        //When
+        viewModel.onTimeChanged(hour, minute)
```

Comment:
nếu ở đây anh nhập giờ > 24 hoặc minute > 60  thì không bị lỗi ah ??

Reply:
Làm sao anh có thể chọn được hour > 24 và minute > 60 khi view nó đang dùng DatePicker ?

Reply:
đây là ViewModel xử lý logic.
anh có biết đầu vào là ntn đâu, nên cần xử lý cho case này chứ

Reply:
`fun onTimeChanged(hour: Int, minute: Int) {
        this.hour = hour
        this.minute = minute
    }`
Vậy thì 
Hàm onTimeChanged nó chỉ thực hiện set 2 giá trị hour và minute không check gì cả. Thì lấy đâu ra logic mà check anh  êi ?

Reply:
thế tức hàm onTimeChanged() viết thiếu hoặc sai logic.
Còn nếu như anh là người viết test mà không cần quan tâm đến người viết code.
Thì chắc chắn sẽ có case là nhập kiểu dữ liệu sai ntn.
em thử check lại xem

Reply:
Cái đó thuộc về đáp án của bài này. 
Trong phạm vi bài training của công ty để thực hiện cách viết UT cho tất cả mọi người.
Và em viết UT đảm bảo đủ điều kiện pass anh ạ.

Reply:
oke